### PR TITLE
feat(markup): no default decoration

### LINKED ISSUE #403: helix pages default markup (html.pre.js) breaking change and refactor
after using our standard markup for a while now, i realized that i would like to remove the `title` and `header` treatment as well as the `image` and `default` classes altogether. 

they seem largely useless and arbitrary, considering the amount of page decoration of the DOM we usually do on the client side.

as this has pretty sizable implications on every helix pages project, we should probably do this sooner than later.

looking at the current `html.pre.js` this would probably also mean that we could remove jquery as a dependency.

i think it would be great to supply a client-sided helper for the projects that relied on the current behaviour.

i am assigning this to @rofe as the remainder of the `html.pre.js` is largely his work.

### PR DESCRIPTION
**Pull request by rofe**
Fixes #403 

- [x] Remove default class assignments (title, image, default) and element wrapping
- [x] Remove jquery dependency
- [x] Provide client-side compatibility/legacy lib

### CODE REVIEW

File: src/html.pre.js [hunk 1]
```diff
@@ -9,7 +9,7 @@
  * OF ANY KIND, either express or implied. See the License for the specific language
  * governing permissions and limitations under the License.
  */
-const jquery = require('jquery');
+// const jquery = require('jquery');
```
Comment (kptdobe) [inline src/html.pre.js:12]: Why not removing the line ?

File: src/html.pre.js [hunk 2]
```diff
@@ -55,30 +56,23 @@ function pre(context) {
     }
     $(document.body).children().wrapAll(div);
   }
+  */
```
Comment (kptdobe) [inline src/html.pre.js:59]: Why not removing the code ?

File: test/unit/html.pre.test.js [hunk 3]
```diff
@@ -34,59 +34,6 @@ describe('Testing pre requirements for main function', () => {
...
-          class: 'customcssclass',
-        },
-      },
-      request,
-    };
-    pre(context);
-
-    const div = dom.window.document.querySelector('div');
-    assert.ok(div.classList.contains('default'));
-    assert.ok(div.classList.contains('customcssclass'));
```
Comment (kptdobe) [inline test/unit/html.pre.test.js:68]: While removing the `default` class is the result of this PR, I think the `customcssclass` should still be there to be able to "type" a page or a section.

File: test/unit/html.pre.test.js [hunk 4]
```diff
@@ -34,59 +34,6 @@ describe('Testing pre requirements for main function', () => {
...
-    assert.ok(div.classList.contains('customcssclass2'));
```
Comment (kptdobe) [inline test/unit/html.pre.test.js:87]: While removing the `default` class is the result of this PR, I think the `customcssclass` or  `customcssclass2` should still be there to be able to "type" a page or a section.

File: src/html.pre.js (hunk 2 above)
Comment (rofe) [reply src/html.pre.js:59]: That was the goal, yes (it's still a draft) 😉

**Comment by rofe** [comment]
The only IT failing now is `run_diff_tests`, which is to be expected. I tested theblog locally side by side and couldn't find any visual discrepancies on the homepage or a post page. But we should probably test a few more Helix Pages sites, especially in conjunction with the new `default-wrap.js` (does anyone have an idea for a better name?)

To test locally:
1. check out a helix pages project
2. `hlx up`, when started hit `^C`
3. `cd .hlx/pages`
4. `rm -rf master`
5. `git clone https://github.com/adobe/helix-pages.git master`
6. `cd master`
7. `git checkout no-decoration`
8. `cd ../../..`
9. `hlx up`

**Comment by kptdobe** [comment]
The `wrapNodes` function was not working when some nodes are text nodes. I could not find a way to make it work. Something wrong with JSDOM. Since we use it in a single place and changed completely the implementation and added corresponding tests.

**Comment by rofe** [comment]
@davidnuescheler it would be great if you could help [test this change](https://github.com/adobe/helix-pages/pull/416#issuecomment-674092761) on as many Helix Pages projects as possible in order to ensure the output is as you would expect it, and to minimize the impact on existing projects.

See https://github.com/adobe/theblog/pull/381 for changes needed in theblog

**Comment by tripodsan** [comment]
@rofe should we add this to `breaking-september` ?

**Comment by rofe** [comment]
Continued in #464

**Pull request closed without merging**